pkgs/build-support/gradle/setup-hook.sh

```diff
@@ -0,0 +1,61 @@
+gradleConfigureHook() {
+    if [ -z "${GRADLE_USER_HOME-}" ]; then
+        GRADLE_USER_HOME="$(mktemp -d)"
+    fi
+    export GRADLE_USER_HOME
+    export TERM=dumb
+    gradleFlagsArray+=(--no-daemon --console plain --init-script "${gradleInitScript:-@init_script@}")
+    if [ -n "${MITM_CACHE_CA-}" ]; then
+        if [ -z "${MITM_CACHE_KEYSTORE-}" ]; then
+            MITM_CACHE_KEYSTORE="$MITM_CACHE_CERT_DIR/keystore"
+            MITM_CACHE_KS_PWD="$(head -c10 /dev/random | base32)"
+            echo y | @jdk@/bin/keytool -importcert -file "$MITM_CACHE_CA" -alias alias -keystore "$MITM_CACHE_KEYSTORE" -storepass "$MITM_CACHE_KS_PWD"
+        fi
+        gradleFlagsArray+=(-Dhttp.proxyHost="$MITM_CACHE_HOST" -Dhttp.proxyPort="$MITM_CACHE_PORT")
+        gradleFlagsArray+=(-Dhttps.proxyHost="$MITM_CACHE_HOST" -Dhttps.proxyPort="$MITM_CACHE_PORT")
+        gradleFlagsArray+=(-Djavax.net.ssl.trustStore="$MITM_CACHE_KEYSTORE" -Djavax.net.ssl.trustStorePassword="$MITM_CACHE_KS_PWD")
+    else
+        gradleFlagsArray+=(--offline)
+    fi
+}
+
+gradle() {
+    command gradle $gradleFlags "${gradleFlagsArray[@]}" "$@"
+}
+
+gradleBuildPhase() {
+    runHook preBuild
+
+    gradle ${enableParallelBuilding:+--parallel} ${gradleBuildTask:-assemble}
```

Comment:
Am I reading this correctly that `--parallel` is not passed by default?

Gradle should probably enable it, same as cmake or meson.

Reply:
parallel building breaks builds pretty often, so I'm not sure this is a good idea

Reply:
Even with gradle? Interesting.

Reply:
done, only openjfx had to have this disabled

Reply:
Interesting.

My experience with Gradle projects so far is that parallel building (or even more experimental features) can generally be assumed to work without much issue, so I think having it enabled is a sane default.